### DISCUSSION TITLE
Filter out warnings

### DISCUSSION THREAD

**Discussion by ndeville** [getpelican/pelican#3046] [Q&A] [answered]
now with 400+ notes, and a URL structure with redundant article slugs, eg:

`books/library`
`apps/library`

or

`books/deep-work`
`home-office/deep-work`

I have many warnings printed while publishing, which is annoying, eg:  

`WARNING  There are 2 items "with slug "books"" with lang  en:`

I have tried:

``` python
import logging
LOG_FILTER = [(logging.WARN, 'There are %s items "%s" with lang %s')]
```

which is note working. 

Note: the `'There are %s items "%s" with lang %s'` string is copied from the source code at https://github.com/getpelican/pelican/blob/d5d792060cd5e4c3ff6da82359589b990fa83f27/pelican/utils.py

What am I doing wrong?

Screenshots: https://notes.nicolasdeville.com/python/library-pelican/#warnings

**Comment by avaris** [comment, marked as answer]
Unfortunately, that's not the template string. It is built dynamically ([here](https://github.com/getpelican/pelican/blob/2a7e691000db4b69b1b4c960141f7c00eff5f06d/pelican/utils.py#L665)) which makes it annoyingly unhelpful for this kind of filtering. Code should probably be adjusted to have a static template, so I would consider this a bug. For now, you can try adding multiple versions of this template

```python
import logging
LOG_FILTER = [
    (logging.WARN, 'There are %s items "%s" with lang %s: \n%s\n%s'),  # 2 item warning
    (logging.WARN, 'There are %s items "%s" with lang %s: \n%s\n%s\n%s'),  # 3 item warning
    (logging.WARN, 'There are %s items "%s" with lang %s: \n%s\n%s\n%s\n%s'),  # 4 item warning
    # ... so on. depending on your site and the kind of warnings you get
]

**Reply by ndeville** [reply]
Thanks @avaris 

Your snippet as is removed 5 out of 10 warnings.

Left with 5x warnings à la:

`There are 3 original (not translated) items with slug "ai":` 

Tried to add:

``` python
    (logging.WARN, 'There are %s original (not translated) items with slug %s: \n%s\n%s'),  # 2 item warning
    (logging.WARN, 'There are %s original (not translated) items with slug %s: \n%s\n%s\n%s'),  # 3 item warning
```

and with quotes:

``` python
    (logging.WARN, 'There are %s original (not translated) items with slug "%s": \n%s\n%s'),  # 2 item warning
    (logging.WARN, 'There are %s original (not translated) items with slug "%s": \n%s\n%s\n%s'),  # 3 item warning
```

to no avail.

🤔

(more detailed here: https://notes.nicolasdeville.com/#avaris-help)

**Reply by avaris** [reply]
Close :) . That one is based on [this](https://github.com/getpelican/pelican/blob/2a7e691000db4b69b1b4c960141f7c00eff5f06d/pelican/utils.py#L694) string. So, should be:

```python
    (logging.WARN, 'There are %s original (not translated) items %s: \n%s\n%s'),  # 2 item warning
    (logging.WARN, 'There are %s original (not translated) items %s: \n%s\n%s\n%s'),  # 3 item warning
```

**Reply by ndeville** [reply]
This works! 😁

Thanks!!

**Reply by bfueldner** [reply]
...or the lazy way:

```
for num in range(2, 30):
    LOG_FILTER.append((logging.WARN, 'There are %s original (not translated) items %s: ' + '\n%s' * num))
    LOG_FILTER.append((logging.WARN, 'There are %s items "%s" with lang %s: ' + '\n%s' * num))
```
